### DISCUSSION TITLE
Usage when FieldArray has depency

### DISCUSSION THREAD

**Discussion by hui21109** [react-hook-form/react-hook-form#2920] [Q&A]
Hi, I am a beginner for using FieldArray.
Today I am working on creating a table that with lower boundary and upper boundary.
like
 
0 <= 3
3 <= 5
...
yup schema is like this

```
yupObject().shape({
          table: yupArray().of(
            yupObject().shape({
              upperBoundary: ....
                }),
            })
          ),
        }),
```

This came out some questions I want to ask.

**Question 1**
since lower boundary start with 0 and always same with the last row's upper boundary.
It is a pure text on UI. Only upper boundary is changeable

Can I assign a name to it but it is a read only value ? I seems not able to do that....
So I pass value to the component of the last row.
It would be great to able to make it become one of the schema...

Question 2
I don't sure if it is the limitation of yup or not.
Can I get the key(index) of the array inside the yup schema ?
Because I need to compare the last row upper boundary is always lower the current one.
And I am not able to do that now, so I can't do it with yup.
It cause another issue. I try to use another function and  setError to specific upper boundary when it onBlur.
But it won't work stable because sometimes yup validate after that function.
So an error is cleared after that.

This question is mainly came out because I want to compare with other row.
It might be solved any one of them have solution.
I have tried many ways but none of them works. Please kindly help me with that.
Thanks a lot.

**Comment by bluebill1049** [comment]
let's do a question at time, perhaps you want to supply a simple codesandbox.
